crates/subspace-farmer/src/bin/subspace-farmer/commands/bench.rs

```diff
@@ -306,3 +306,202 @@ fn get_size(path: impl AsRef<Path>) -> std::io::Result<u64> {
     }
     Ok(size)
 }
+
+#[cfg(test)]
+#[tokio::test]
+async fn test_dsn_sync() {
+    use std::ops::Range;
+
+    use subspace_farmer::{PieceIndexHashNumber, SyncResult};
+    use subspace_networking::libp2p::multiaddr::Protocol;
+    use subspace_networking::libp2p::{identity, Multiaddr, PeerId};
+
+    let seeder_max_plot_size = 20 * 1024 * 1024 / PIECE_SIZE as u64; // 20M
+    let syncer_max_plot_size = 2 * 1024 * 1024 / PIECE_SIZE as u64; // 20M
+    let request_pieces_size = 20;
+
+    let seeder_base_directory = TempDir::new().unwrap();
+    let mut seeder_multiaddr: Multiaddr = "/ip4/127.0.0.1/tcp/40000".parse().unwrap();
```

Comment:
Better to use random port and catch correct one afterwards, this approach will likely cause intermittent issues in CI and otherwise